rust/azure_iot_operations_mqtt/src/aio/cloud_event.rs

```diff
@@ -631,4 +631,495 @@ mod tests {
     fn test_cloud_event_validate_invalid_spec_version() {
         CloudEventFields::Id.validate("id", "0.0").unwrap_err();
     }
+
+    #[test]
+    fn test_try_from_publish_properties_success() {
+        let user_properties = vec![
+            ("id".to_string(), "test-event-123".to_string()),
+            ("source".to_string(), "aio://sensor/temperature".to_string()),
+            ("specversion".to_string(), "1.0".to_string()),
+            ("type".to_string(), "ms.aio.rpc.request".to_string()),
+            ("subject".to_string(), "sensor-01".to_string()),
+            ("time".to_string(), "2025-12-11T00:00:00Z".to_string()),
+            (
+                "dataschema".to_string(),
+                "https://example.com/schema".to_string(),
+            ),
+        ];
+
+        let publish_properties = PublishProperties {
+            user_properties,
+            content_type: Some("application/json".to_string()),
+            ..Default::default()
+        };
+
+        let cloud_event = CloudEvent::try_from(&publish_properties).unwrap();
+
+        assert_eq!(cloud_event.id, "test-event-123");
+        assert_eq!(cloud_event.source, "aio://sensor/temperature");
+        assert_eq!(cloud_event.spec_version, "1.0");
+        assert_eq!(cloud_event.event_type, "ms.aio.rpc.request");
+        assert_eq!(cloud_event.subject, Some("sensor-01".to_string()));
+        assert_eq!(
+            cloud_event.data_schema,
+            Some("https://example.com/schema".to_string())
+        );
+        assert_eq!(
+            cloud_event.data_content_type,
+            Some("application/json".to_string())
+        );
+        assert!(cloud_event.time.is_some());
+    }
+
+    #[test]
+    fn test_try_from_publish_properties_minimal_required_fields() {
+        let user_properties = vec![
+            ("id".to_string(), "minimal-event".to_string()),
+            ("source".to_string(), "aio://device/001".to_string()),
+            ("specversion".to_string(), "1.0".to_string()),
+            ("type".to_string(), "ms.aio.rpc.request".to_string()),
+        ];
+
+        let publish_properties = PublishProperties {
+            user_properties,
+            ..Default::default()
+        };
+
+        let cloud_event = CloudEvent::try_from(&publish_properties).unwrap();
+
+        assert_eq!(cloud_event.id, "minimal-event");
+        assert_eq!(cloud_event.source, "aio://device/001");
+        assert_eq!(cloud_event.spec_version, "1.0");
+        assert_eq!(cloud_event.event_type, "ms.aio.rpc.request");
+        assert_eq!(cloud_event.subject, None);
+        assert_eq!(cloud_event.data_schema, None);
+        assert_eq!(cloud_event.data_content_type, None);
+        assert_eq!(cloud_event.time, None);
+    }
+
+    #[test]
+    fn test_try_from_publish_properties_missing_required_fields() {
```

Comment:
```suggestion
    fn test_try_from_publish_properties_missing_source() {
```
to match the other test names